### ISSUE TITLE
Private messages

### ISSUE THREAD

**Issue by lxwang42** [DimiMikadze/orca#21]
any plan to implement private message feature?
it would be nice to allow users to exchange some whispers. 😄

**Comment by DimiMikadze** [comment]
Hi @lxwang42 thanks for the issue, i agree that would be quite nice feature to have 🙂

Unfortunately I don't have time to implement it, but if you or anyone else is interested to make a PR, I'll be happy to merge it. Just before starting working on it we have to discuss how you are going to implement it, how UI should look etc.

**Comment by rachid-aachich** [comment]
@DimiMikadze @lxwang42  That sounds like something cool to work on, i'd love to start it once i get the time, any methods, libraries or ideas to suggest? websockets, long-polling?

**Comment by DimiMikadze** [comment]
Hi @rachid-aachich for real time functionality we need to use GraphQL Subscriptions: https://www.howtographql.com/react-apollo/8-subscriptions/

Create new mongoose model  `Message`, new resolver `message`.

Regarding UI I think we can create separate page for it, like Twitter: https://twitter.com/messages

Let me know when You'll be ready to start working on it and we can discuss it in details ✌️

**Comment by leventdeniz** [comment]
How would you like it if there would be an extra branch for this feature so we all can push smaller changes to be merged into that branch? This might speed up the development if one person doesn't have the time to do it all at once.

**Comment by DimiMikadze** [comment]
Hi @leventdeniz currently i'm working on integration of realtime functionality into repository ( GraphQL subscriptions ). I think by end of the following week We'll have notifications realtime. After that it will be much easier to implement other real time functionalities e.g. Private Messages. I'll write my progress in here ✌️

**Comment by DimiMikadze** [comment]
Regarding my progress, I've integrated GraphQL subscriptions into the repository, you can find them on master branch. Regarding Private messages, We have started working on it, this is the branch that contains all the changes for it: https://github.com/udilia/create-social-network/tree/implement-private-messages Currently we have built all needed queries and mutations on backend side and Messages UI and functionality is in progress. I think we'll finish them after 2 or 3 weeks, since there is a lot of work to do to finish them and i have a vacation 🙃Feel free to contribute any way you like ✌️

**Comment by leventdeniz** [comment]
Are there any features you want to be implemented? @DimiMikadze

**Comment by leventdeniz** [comment]
Fixed some things and noticed that it is possible to send empty messages. I'd also love a textarea for the message input that automatically expands.

**Comment by DimiMikadze** [comment]
Hi Guys, since many people are already working on this PR and there's still many thing to do to finish it, let me divide it as a sub tasks, so we can assign ourselves to individual sub tasks and start working on them. That way we'll avoid merge conflicts and we'll have clear a goal to accomplish.
Please give me 2 days for it, since I'm on vacation right now and thanks for contributing! ✌️

**Comment by DimiMikadze** [comment]
I've added 4 sub issues, I think if we'll finish them we'll be able to close this issue, if you think we have to do something else feel free to fill an issue.

If you decide to work on any of them, please first assign it to yourself so everybody knows you are working on it and if you'll have any questions or issue descriptions are not clear enough please ask first, so we don't waste time.

We'll work on redesigning of messages page later, first let's finish its core functionality :) ✌️

**Comment by DimiMikadze** [comment]
Hey guys, Messenger has been merged to master 🎉 you can test it on demo website 🙂Code will need some refactoring and UI will need some changes, which I'll do in the following week. For now let's test it and create issues if we'll find bugs around it ✌️

**Comment by reesecode** [comment]
I actually pulled it down soon as I saw the pr go through! 

Ive got a fresh bug for ya, I'll try to see how to reproduce and make a issue. has to do with new messages and getting "TypeError: chatUser is null" when visiting http://localhost:3000/messages/new at times
